libs/platform/ebpf_tracelog.c

```diff
@@ -0,0 +1,1059 @@
+        break;                                                                           \
+    CASE_ERROR:                                                                          \
+        _EBPF_LOG_MESSAGE_GUID(trace_level, KEYWORD_ERROR, message, guid);               \
+        break;                                                                           \
+    CASE_EPOCH:                                                                          \
+        _EBPF_LOG_MESSAGE_GUID(trace_level, KEYWORD_EPOCH, message, guid);               \
+        break;                                                                           \
+    CASE_CORE:                                                                           \
+        _EBPF_LOG_MESSAGE_GUID(trace_level, KEYWORD_CORE, message, guid);                \
+        break;                                                                           \
+    CASE_LINK:                                                                           \
+        _EBPF_LOG_MESSAGE_GUID(trace_level, KEYWORD_LINK, message, guid);                \
+        break;                                                                           \
+    CASE_MAP:                                                                            \
+        _EBPF_LOG_MESSAGE_GUID(trace_level, KEYWORD_MAP, message, guid);                 \
+        break;                                                                           \
+    CASE_PROGRAM:                                                                        \
+        _EBPF_LOG_MESSAGE_GUID(trace_level, KEYWORD_PROGRAM, message, guid);             \
+        break;                                                                           \
+    CASE_API:                                                                            \
+        _EBPF_LOG_MESSAGE_GUID(trace_level, KEYWORD_API, message, guid);                 \
+        break;                                                                           \
+    CASE_PRINTK:                                                                         \
+        _EBPF_LOG_MESSAGE_GUID(trace_level, KEYWORD_PRINTK, message, guid);              \
+        break;                                                                           \
+    CASE_NATIVE:                                                                         \
+        _EBPF_LOG_MESSAGE_GUID(trace_level, KEYWORD_NATIVE, message, guid);              \
+        break;                                                                           \
+    default:                                                                             \
+        break;                                                                           \
+    }
+__declspec(noinline) void ebpf_log_message_guid(
+    ebpf_tracelog_level_t trace_level,
+    ebpf_tracelog_keyword_t keyword,
+    _In_z_ const char* message,
+    _In_ const GUID guid)
```

Comment:
```suggestion
    GUID guid)
```
It might actually be worth making all these be `GUID*` (and leaving the `_In_ const`) instead of `GUID`. That may even improve perf.

Reply:
Agree to change the existing implementation: even though `TraceLoggingGuid` will anyways make a copy, pushing a pointer up to it would save some.